plugins/modules/cloud/misc/proxmox_disk.py

```diff
@@ -0,0 +1,719 @@
+    api_password: secret
+    vmid: 101
+    disk: scsi4
+    state: absent
+'''
+
+RETURN = '''
+vmid:
+  description: The VM vmid.
+  returned: success
+  type: int
+  sample: 101
+msg:
+  description: A short message
+  returned: always
+  type: str
+  sample: "Disk scsi3 created in VM 101"
+'''
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback
+from ansible.errors import AnsibleOptionsError
+from ansible_collections.community.general.plugins.module_utils.proxmox import (proxmox_auth_argument_spec,
+                                                                                ProxmoxAnsible)
+from re import compile, match, sub
+from time import sleep
+
+
+def disk_conf_str_to_dict(config_string):
+    config = config_string.split(',')
+    storage_volume = config.pop(0).split(':')
+    config.sort()
+    storage_name = storage_volume[0]
+    volume_name = storage_volume[1]
+    config_current = dict(
+        volume='%s:%s' % (storage_name, volume_name),
+        storage_name=storage_name,
+        volume_name=volume_name
+    )
+
+    for option in config:
+        k, v = option.split('=')
+        config_current[k] = v
+
+    return config_current
+
+
+class ProxmoxDiskAnsible(ProxmoxAnsible):
+    create_update_fields = [
+        'aio', 'backup', 'bps_max_length', 'bps_rd_max_length', 'bps_wr_max_length',
+        'cache', 'cyls', 'detect_zeroes', 'discard', 'format', 'heads', 'import_from', 'iops', 'iops_max',
+        'iops_max_length', 'iops_rd', 'iops_rd_max', 'iops_rd_max_length', 'iops_wr', 'iops_wr_max',
+        'iops_wr_max_length', 'iothread', 'mbps', 'mbps_max', 'mbps_rd', 'mbps_rd_max', 'mbps_wr', 'mbps_wr_max',
+        'media', 'queues', 'replicate', 'rerror', 'ro', 'scsiblock', 'secs', 'serial', 'shared', 'snapshot',
+        'ssd', 'trans', 'werror', 'wwn'
+    ]
+    supported_bus_num_ranges = dict(
+        ide=range(0, 4),
+        scsi=range(0, 31),
+        sata=range(0, 6),
+        virtio=range(0, 16),
+        unused=range(0, 256)
+    )
+
+    def sanitize_params(self):
+        # Sanitize parameters dictionary:
+        # - Remove not defined args
+        # - Ensure True and False converted to int.
+        # - Remove unnecessary parameters
+        params = dict((k, v) for k, v in self.module.params.items() if v is not None and k in self.create_update_fields)
+        params = dict((k, int(v)) for k, v in params.items() if isinstance(v, bool))
+        return params
+
+    def create_disk(self, disk, vmid, vm, vm_config):
+        # Would not replace existing disk without 'force' flag
+        force_replace = self.module.params['force_replace']
+        if not force_replace and disk in vm_config:
+            return False
+
+        params = self.sanitize_params()
+        import_string = params.pop('import_from', None)
+        if import_string and self.module.params["size"] != "0":
+            raise AnsibleOptionsError('Only size=0 is acceptable with <import_from> parameter.')
```

Comment:
Maybe it would make more sense to make the parameters `import_from` and `size` mutually exclusive in the module, and add `storage:0` to `config_str` inside this `if` block.

Reply:
Great idea! I havent found a supported method to achieve this when wrote this code. Thank you.